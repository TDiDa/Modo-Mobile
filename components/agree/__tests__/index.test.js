import React from 'react';
import { render } from 'enzyme';
import Agree from '..';

describe('Agree', () => {
  it('renders correctly', () => {
    const wrapper = render(<Agree>Default</Agree>);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a simple checked agree', () => {
    const wrapper = render(<Agree value>Default</Agree>);
    expect(wrapper.hasClass('m-agree-checked')).toBe(true);
  });

  it('renders a disabled agree', () => {
    const wrapper = render(<Agree disabled>Default</Agree>);
    expect(wrapper.hasClass('m-agree-disabled')).toBe(true);
  });
});
