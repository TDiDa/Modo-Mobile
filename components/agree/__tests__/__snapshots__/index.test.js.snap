// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Agree renders correctly 1`] = `
<div
  class="m-agree"
>
  <div
    class="m-agree-icon"
  >
    <i
      class="m-icon m-icon-check-circle"
    >
      <svg
        aria-hidden="true"
        class=""
        fill="currentColor"
        height="1em"
        width="1em"
      >
        <use
          href="#icon-check-circle"
        />
      </svg>
    </i>
  </div>
  <div
    class="m-agree-content"
  >
    Default
  </div>
</div>
`;
